src/compiler/emitter.ts

```diff
@@ -3445,10 +3455,15 @@ namespace ts {
          * makeUniqueName are guaranteed to never conflict.
          * If `optimistic` is set, the first instance will use 'baseName' verbatim instead of 'baseName_1'
          */
-        function makeUniqueName(baseName: string, checkFn: (name: string) => boolean = isUniqueName, optimistic?: boolean): string {
+        function makeUniqueName(baseName: string, checkFn: (name: string) => boolean = isUniqueName, optimistic?: boolean, scoped?: boolean): string {
             if (optimistic) {
                 if (checkFn(baseName)) {
-                    generatedNames.set(baseName, true);
+                    if (scoped) {

```

Comment:
Anything that goes in `generatedNames` is already reserved throughout the entire generated output, so scoping these doesn't seem to make any sense. Why was this needed?

Reply:
`scoped` causes it to go into `reservedNames` instead of `generatedNames`; this allows, eg, `T_1` to be reused in adjacent scopes.